src/Progress/index.tsx

```diff
@@ -163,8 +163,8 @@ export const ProgressItem = ({
   </div>
 );
 
-export const Progress: React.ComponentType<{}> & { Item: typeof ProgressItem } = ({ children }) => (
-  <ProgressSection>{children}</ProgressSection>
-);
+export const Progress: React.ComponentType<React.PropsWithChildren<{}>> & {
+  Item: typeof ProgressItem;
```

Comment:
Same here (not PR related, just a general comment)
...are we extending React Component types to also have an `Item` property? 
seems odd and unidiomatic, but there's probably a reasoning that could just used being documented and spelled out for other devs + users.